[BUG] Initializing varied length array in src/Enzo/io/EnzoInitialHdf5.cpp

**Describe the bug**
When building Enzo-e on ARM MacOS, the system runs into a variable-sized object may not be initialized error. It occurs at EnzoInitialHdf5.cpp, line 122.

**To Reproduce**
Steps to reproduce the behavior:
1. Follow the [Getting Started](https://enzo-e.readthedocs.io/en/main/tutorial/getting_started.html) guide using MacOS 14.5 on ARM.
2. Run `make` after configuring cmake build options for Enzo-e/Cello

**Error messages**


```
error: variable-sized object may not be initialized
  int count_messages[max_level_ + 1] = {};
                     ^~~~~~~~~~~~~~
1 warning and 1 error generated.
make[2]: *** [src/Enzo/io/CMakeFiles/Enzo_io.dir/EnzoInitialHdf5.cpp.o] Error 1
make[1]: *** [src/Enzo/io/CMakeFiles/Enzo_io.dir/all] Error 2
make: *** [all] Error 2
```

**Expected outcome**
Should have no errors to ensure successful installation.

**Version information (please complete the following information):**
 - OS: MacOS 14.5
 - Charm++ version: 7.0.0
 - Compiler version: clang 15.0.0

Uploading `.cmake` files seem to not be supported here. Here's a copy of the content:
```
# Machine configs are included twice.
# First to set compilers, paths, default options.
# Second to set dependent options (e.g., also depending on defaults set in the main CMakeLists.txt)

# This is the configuration for a generic macOS installation using clang and
# gfortran
# - this was tested on a clean installation of version 12.5 of macOS on an
#   Intel-based MacBook Pro on July 24, 2022
# - historically there were challenges associated with installing Charm++ and
#   Cello/Enzo-E on Macs, but these issues seem to have been implicitly
#   resolved in each project's transition to using CMake for their
#   build-systems
#
# In this succesful installation:
# - Homebrew was used to install:
#       gfortran, hdf5, libpng, boost, and CMake
#   If you install this software through other channels, you may need to
#   provide CMake with hints about the locations of these packages (see the
#   main documentation for details).
# - the netlrts backend was used for Charm++
#
# If you're having problems with your installation:
# - did you make sure gfortran was installed before building Charm++?


if(NOT __processedUserDefaults)

  message(STATUS "Loading machine configuration for MacOS with clang and gfortran.\n"
    "If you encounter installation problems, please read the comments at the "
    "top of config/darwin_clang.cmake"
  )

  # Setting compilers
  set(CMAKE_CXX_COMPILER clang++ CACHE STRING "")
  set(CMAKE_C_COMPILER clang CACHE STRING "")
  set(CMAKE_Fortran_COMPILER gfortran CACHE STRING "")
  set(CMAKE_Fortran_FLAGS "-ffixed-line-length-132" CACHE STRING "Default Fortran flags")

  # these flag(s) are currently only used when using openmp-simd optimizations
  # (to specify available/prefered instruction sets).
  # This particular value tells the compiler to optimize the code for the
  # instruction set of the machine used to compile the code.
  set(CONFIG_ARCH_FLAGS "-march=native")

  # add flag to unroll loops (this flag would also be enabled anyways when
  # OPTIMIZE_FP=TRUE)
  set(CMAKE_C_FLAGS "-funroll-loops" CACHE STRING "Default C flags")
  set(CMAKE_CXX_FLAGS "${CMAKE_C_FLAGS}" CACHE STRING "Default C++ flags")

  # Setting package paths (e.g., Grackle) - (meant for personal machine files)

  # Mark done
  set(__processedUserDefaults ON)

else()

  if (USE_DOUBLE_PREC)
    string(APPEND CMAKE_Fortran_FLAGS " -fdefault-real-8 -fdefault-double-8")
  endif()

endif()

```

**Additional context**
Found a temporary fix by using std::vector<int> instead of integer array. That is, replacing line 122 with `std::vector<int> count_messages(max_level_ + 1);`. It does the job, but I do not know if it is the best way to do it.

--- a/src/Enzo/io/EnzoInitialHdf5.cpp
+++ b/src/Enzo/io/EnzoInitialHdf5.cpp
@@ -10,6 +10,7 @@
 
 #include <chrono>
 #include <thread>
+#include <vector>
 
 #define CHECK_COSMO_PARAMS true
 
@@ -119,7 +120,7 @@ void EnzoInitialHdf5::enforce_block
   int min_level = cello::hierarchy()->min_level();
 
   // Maintain running count of messages sent
-  int count_messages[max_level_ + 1] = {};
+  std::vector<int> count_messages(max_level_ + 1, 0);
 
   // Read in Field files
   for (size_t index=0; index<field_files_.size(); index++) {

